proposal/color-4-new-spaces-js.d.ts.md

```diff
@@ -0,0 +1,931 @@
+  `options.lightness`.
+
+* Let `c` be the result of [parsing a channel value] with value `options.c`.
+
+* Let `h` be the result of [parsing a channel value] with value `options.h`.
+  
+* If `options.alpha` is not set, let `alpha` be `1`. Otherwise, let `alpha` be
+    the result of [parsing a channel value] with value `options.alpha`.
+
+* If `options.space` equals `lch`, set [`internal`] to the result of
+  [`lch(lightness a b / alpha)`].
+
+* Otherwise, if `options.space` equals `oklch`, set [`internal`] to the result
+  of [`oklch(lightness a b / alpha)`].
+
+```ts
+constructor(options: {
+  lightness: number | null;
+  chroma: number | null;
+  hue: number | null;
+  alpha?: number | null;
+  space: ColorSpaceLCH;
+});
+```
+
+#### Predefined RGB Channel Constructor
+
+Create a new SassColor in a color space with RGB channels -- `srgb`,
+`srgb-linear`, `display-p3`, `a98-rgb`, and `prophoto-rgb`. `rgb` is supported
+through the modifed [RGB Constructor].
+
+* Let `red` be the result of [parsing a channel value] with value `options.red`.
+
+* Let `green` be the result of [parsing a channel value] with value
+  `options.green`.
+
+* Let `blue` be the result of [parsing a channel value] with value
+  `options.blue`.
+  
+* If `options.alpha` is not set, let `alpha` be `1`. Otherwise, let `alpha` be
+    the result of [parsing a channel value] with value `options.alpha`.
+
+* Let `space` be the unquoted string value of `options.space`.
+
+* Set [`internal`] to the result of [`color(space red green blue / alpha)`].
+
+[`color(space red green blue / alpha)`]: ./color-4-new-spaces.md#color-1
+[RGB Constructor]: #rgb-constructor
+
+```ts
+constructor(options: {
+  red: number | null;
+  green: number | null;
+  blue: number | null;
+  alpha?: number | null;
+  space: Omit<ColorSpaceRGB, 'rgb'>;
+});
+```
+
+#### XYZ Channel Constructor
+
+Create a new SassColor in a color space with XYZ channels -- `xyz`, `xyz-d50`,
+and `xyz-d65`.
+
+* Let `x` be the result of [parsing a channel value] with value `options.x`.
+
+* Let `y` be the result of [parsing a channel value] with value `options.y`.
+
+* Let `z` be the result of [parsing a channel value] with value `options.z`.
+  
+* If `options.alpha` is not set, let `alpha` be `1`. Otherwise, let `alpha` be
+    the result of [parsing a channel value] with value `options.alpha`.
+
+* Let `space` be the unquoted string value of `options.space`.
+
+* Set [`internal`] to the result of [`color(space x y z / alpha)`].
+
+```ts
+constructor(options: {
+  x: number | null;
+  y: number | null;
+  z: number | null;
+  alpha?: number | null;
+  space: ColorSpaceXYZ;
+});
+```
+
+### Modified Legacy Color Constructors
+
+These will replace the [existing constructors] for legacy colors.
+
+#### HSL Constructor
+
+Create a new SassColor in the `hsl` color space.
+
+* If `options.alpha` is `null` and `options.space` is not set, emit a
+  deprecation warning named `null-alpha`.
```

Comment:
@jamesnw I want to be sure I understand the intent behind the "and `options.space` is not set" clauses in these modifications to the legacy color constructors. Does that mean if `alpha: null` and `space: 'hsl'` are used, then `alpha` will be interpreted as `"none"` as per [Parsing a Channel Value](https://github.com/sass/sass/blob/main/proposal/color-4-new-spaces-js.d.ts.md#parsing-a-channel-value)?

Reply:
That's right. If you use the new constructor format, you get the new semantics.